src/libraries/System.Data.Common/src/System/Data/Common/DataStorage.cs

```diff
@@ -557,6 +564,7 @@ protected void SetNullStorage(BitArray nullbits)
         /// Types like "System.Data.SqlTypes.SqlString" will load because they are in the same assembly as this code
         /// Types like "System.Numerics.BigInteger" won't load because they are not special and not same assembly as this code
         /// </remarks>
+        [RequiresUnreferencedCode(DataSet.RequiresUnreferencedCodeMessage)]
```

Comment:
Message here will be confusing if we use the general one right? Seems like the unsafe part from here is just that we are calling Type.GetType on a string.